internal/services/mysql/mysql_flexible_server_resource.go

```diff
@@ -355,6 +360,13 @@ func resourceMysqlFlexibleServerCreate(d *pluginsdk.ResourceData, meta interface
 		}
 	}
 
+	storageSettings := expandArmServerStorage(d.Get("storage").([]interface{}))
+	if storageSettings != nil {
+		if storageSettings.Iops != nil && *storageSettings.AutoIoScaling == servers.EnableStatusEnumEnabled {
+			return fmt.Errorf("`iops` can not be set if `io_scaling_enabled` is set to true")
+		}
+	}
+
```

Comment:
@stephybun

What do you think about this check that i have added in the `resourceMysqlFlexibleServerCreate` function. Should i add it to the `resourceMysqlFlexibleServerUpdate` function as well?

Reply:
Adding this check to the update will be problematic because `iops` is a `Computed` property. 

This means it will always have a value and cannot be cleared. Adding this check to the update would prevent users from being able to switch over to `io_scaling_enabled` so I think we should omit it here for now.